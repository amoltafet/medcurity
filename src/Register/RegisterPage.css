.registerbg {
    background-image: url("../Login/loginbg_small.png");
    height: 100vh;
    width: 100%;
    background-position: center;
    background-repeat: no-repeat;
    position: fixed;
    
}

@media screen and (min-width: 1000px) {
    .registerbg {
        background-image: url("../Login/loginbg_medium.png");
        height: 100vh;
        width: 100%;
        background-position: center;
        background-repeat: no-repeat;
        position: fixed;
    }
  }

@media screen and (min-width: 1400px) {
    .registerbg {
        background-image: url("../Login/loginbg.png");
        height: 100vh;
        width: 100%;
        background-position: center;
        background-repeat: no-repeat;
        position: fixed;
    }
  }

.register_registerCard {
    border: white;
    background-color: transparent;
    width: 70%;
}

.register_pageHeaderText {
    color: white;
    text-shadow: 2px 2px 5px black;
    font-size: 3vmax;
}

.register_columnDivder {
    margin-top: 10%;
    animation: register_upwardsFadeIn 0.5s linear;
}


.register_formColumn {
    border-radius: 25px;
    padding-top: 2%;
    padding-bottom: 2%;
    
}

.register_formColumn:hover {
    padding-top: 2%;
    padding-bottom: 2%;
    animation: register_hoverFade 0.5s linear;
    background-color: #ffffff1f; 
}

.register_h3{
    color: white;
    text-align: center;
    width: 75%;
    margin-bottom: 2%;
    text-shadow: 2px 2px 2px black;
}

.register_p{
    color: white;
    text-align: center;
    width: 75%;
}

.register_registerResponse {
    margin-top: 2%;
    font-family: Lato,sans-serif;
    font-size: 1vmax;
    color: white;
    width: 72%;
    background-color: #0000003f;
}

.register_Form{
    width: 75%;
}

.register_button {
    color: white;
    background-color: #cc3333;
    font-family: Lato,sans-serif;
    font-size: 1.3vmax;
    text-align: center;
    width: 72%;
    height: 20%;
    margin-bottom: 3%;
}

.register_button:hover {
    animation: register_buttonHover .3s linear
}

@keyframes register_hoverFade { 
    from { background-color: #ffffff00; } 
    to { background-color: #ffffff1f; }
}

@keyframes register_upwardsFadeIn { 
    from { margin-top:15%; opacity:0; } 
    to { margin-top:10%; opacity:1; } 
}

@keyframes register_buttonHover { 
    0% { font-size: 1.3vmax; } 
    50% { font-size: 1.32vmax; } 
    0% { font-size: 1.3vmax; } 
}