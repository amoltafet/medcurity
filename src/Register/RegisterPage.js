import {Button, Image, Form, Card} from 'react-bootstrap'
import React from 'react';
import { useState } from "react";
import Axios from "axios"
import './RegisterPage.css';
import 'bootstrap/dist/css/bootstrap.min.css'
import { useNavigate } from 'react-router-dom';

/**
* Creates and displays the main login page. 
* @return {RegisterPage}
*/

export default function RegisterPage()
{
  Axios.defaults.withCredentials = true;

  const [message, setMessage] = useState("")
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const navigate = useNavigate();

  const register = () => {
    console.log('REGISTERING', email, password)
    Axios.post("http://localhost:3002/users/register",
    { 
      email: email,
      password: password
    }).then((response) => 
    {
      console.log("response.data =", response.data)
      if (response.data === true)
      {
        console.log("A new user!")
        navigate('/dash');
      }
      else if (response.data === false)
      {
        console.log("Already has account!")
        setMessage('This email is already in use! Please go back to the login page by clicking the Medcurity logo.')
      }
    }).catch(error => console.error(`Error ${error}`));
  };

  const login = () => {
    navigate('/');
  };
  
      return (
          <>
          <Form className="registerbg img-fluid">
            <a href="/">
            <Image className="MedcurityLogo justify-content-bottom" variant="top" src="/Medcurity_Logo.png" alt="" />
            </a>
              <Card className="registerCard">
                <Card.Text className="header">Medcurity Learn Security</Card.Text>
                <Card.Text className="body">
                  After creating your account, you'll be able to access the various HIPAA learning content.
                </Card.Text>
              </Card>
            <Form className="emailAndPass">
              <Form.Text className="registerHeader">Create an account for Medcurity Learn Security</Form.Text>
                <Form.Group className="email" controlId="formEmail">
                  <Form.Control 
                    type="email" 
                    placeholder="Email" 
                    onChange={ (e) => 
                    {
                      setEmail(e.target.value);
                    }}/>
                </Form.Group>
                <Form.Group className="pass" controlId="formPassword">
                  <Form.Control 
                    type="password" 
                    placeholder="Password"
                    onChange={ (e) => 
                    {
                      setPassword(e.target.value);
                    }}/> 
                  <Button className="loginRegisterBtn" variant="secondary" type="button" onClick={login}>Login</Button>
                  <Button className="createButton" variant="secondary" type="button" onClick={register}>Register</Button>
                </Form.Group>
                <Form.Text className="registerMessage">{message}</Form.Text>
            </Form>
          </Form>
          </>
      );
}

