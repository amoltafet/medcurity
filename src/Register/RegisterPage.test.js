import renderer from 'react-test-renderer';
import RegisterPage from './RegisterPage';
import React from 'react';

test('renders LoginPage', () => {
  const tree = renderer.create(<RegisterPage />).toJSON();
  expect(tree).toMatchSnapshot();
});