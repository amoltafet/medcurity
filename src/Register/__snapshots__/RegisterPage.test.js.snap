// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders RegisterPage 1`] = `
<form
  className="registerbg img-fluid"
>
  <img
    alt=""
    className="medcurity_logo justify-content-bottom"
    src="/triangle_logo.png"
    variant="top"
  />
  <form
    className="register_columnDivder"
  >
    <div
      class="row justify-content-md-center"
    >
      <div
        class="col-xs-5 col-md-5"
      >
        <div
          class="register_formColumn row justify-content-center"
        >
          <h3
            class="register_h3"
          >
            Create an Account
          </h3>
          <p
            class="register_p"
          >
            All you need is an email and password to create an account. If you recieved an invitation, please use the same email you receieved it.
          </p>
          <div
            className="register_Form form-group"
          >
             
            <input
              className="form-control"
              id="formEmail"
              onChange={[Function]}
              placeholder="Email"
              type="email"
            />
             
          </div>
          <p />
          <div
            className="register_Form form-group"
          >
             
            <input
              className="form-control"
              id="formPassword"
              onChange={[Function]}
              placeholder="Password"
              type="password"
            />
             
          </div>
          <p
            className="register_registerResponse"
          >
            
          </p>
          <button
            className="register_button btn btn-secondary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Create Account
          </button>
          <button
            className="register_button btn btn-secondary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Back to Login Page
          </button>
        </div>
      </div>
    </div>
  </form>
</form>
`;
